### ISSUE TITLE
Generated API Client not respecting basePath when passing baseURL

### ISSUE THREAD

**Issue by pseudo-su** [deepmap/oapi-codegen#126]
Possibly related: https://github.com/deepmap/oapi-codegen/issues/55 (but more specifically issues with the generated client)

Using this: https://swagger.io/docs/specification/api-host-and-base-path/
<img width="740" alt="Screen Shot 2020-02-04 at 1 40 55 pm" src="https://user-images.githubusercontent.com/4339582/73708998-03838f80-4754-11ea-9dfd-b448716b9a3a.png">

## Problem

Given a swagger spec that looks like this:
```yaml
openapi: 3.0.0
servers:
  - url: https://my-server.com/v1/some/base/path
paths:
  /things:
    get:
      operationId: getThings
      responses:
        "200":
          content:
            application/json:
              schema:
                type: object
```

And an api client initialized like this:
```go
client := NewClientWithResponses("https://my-server.com/v1/some/base/path")

resp, err := client.GetThingsWithResponse(context.TODO())
```

The HTTP request sent by the generated client isn't what I expected it to be

Expected request: `GET https://my-server.com/v1/some/base/path/things`
Actual request: `GET https://my-server.com/things`

## Investigation

Looking through the generated code, this seems to be the cause of my problem (comments added for explaination):

```golang
// when server == "https://my-server.com/v1/some/base/path"
queryUrl, err := url.Parse(server)
if err != nil {
	return nil, err
}
// passing the absolute path "/things" overwrites the existing basePath "/v1/some/base/path"
queryUrl, err = queryUrl.Parse("/things")
if err != nil {
	return nil, err
}
```
I think according to the docs [here](https://golang.org/pkg/net/url/#URL.Parse) it should be `queryUrl.Parse("things")` instead so it doesn't overwrite the basePath in the baseURL. But we would also have to make sure to append a "/" to the end of the baseURL if it isn't present

<img width="851" alt="Screen Shot 2020-02-04 at 5 03 39 pm" src="https://user-images.githubusercontent.com/4339582/73718345-54a17c80-4770-11ea-91dd-a8319f4c0db3.png">

repl.it link: https://repl.it/repls/ColorlessNumbArchitects
<img width="1092" alt="Screen Shot 2020-02-04 at 5 22 28 pm" src="https://user-images.githubusercontent.com/4339582/73719252-ef9b5600-4772-11ea-8fdc-0bc940f41f93.png">

**Comment by pseudo-su** [comment]
It looks like this is fixed as of `v1.3.6` so I'm closing this issue, thankyou 🙏

**Comment by pseudo-su** [comment]
Just got around to checking this properly and I don't think this is completely fixed.

I have a test case that should be enough to reproduce the behavior

```
func TestTemp(t *testing.T) {
	mockClient := &MockClient{}

	client1, err := pkg.NewClient(
		"https://my-api.com/some-base-url/v1/",
		pkg.WithHTTPClient(mockClient),
	)
	assert.NoError(t, err)

	client2, err := pkg.NewClient(
		"https://my-api.com/some-base-url/v1",
		pkg.WithHTTPClient(mockClient),
	)
	assert.NoError(t, err)

	client3, err := pkg.NewClient(
		"",
		pkg.WithHTTPClient(mockClient),
		pkg.WithBaseURL("https://my-api.com/some-base-url/v1"),
	)
	assert.NoError(t, err)

	client4, err := pkg.NewClient(
		"",
		pkg.WithHTTPClient(mockClient),
		pkg.WithBaseURL("https://my-api.com/some-base-url/v1/"),
	)
	assert.NoError(t, err)

	expectedURL := "https://my-api.com/some-base-url/v1/"

	assert.Equal(t, expectedURL, client1.Server)
        // FAILURE! client2.Server doesn't end with a trailing `/`
        // so the relative paths get messed up for operations
	assert.Equal(t, expectedURL, client2.Server)
	assert.Equal(t, expectedURL, client3.Server)
	assert.Equal(t, expectedURL, client4.Server)
}
```
